clang/lib/CConv/ConstraintResolver.cpp

```diff
@@ -245,10 +229,10 @@ CVarSet
           !(SubTypE->isFunctionType() || SubTypE->isArrayType() ||
             SubTypE->isVoidPointerType()) &&
           !isCastSafe(TypE, SubTypE)) {
-        std::string Rsn = "Cast from " + SubTypE.getAsString() +  " to " +
-                          TypE.getAsString();
-        constraintAllCVarsToWild(CVs, Rsn, IE);
-        return getInvalidCastPVCons(E);
+        CVarSet WildCVar = getInvalidCastPVCons(IE);
+        constrainConsVarGeq(CVs, WildCVar, CS, nullptr, Safe_to_Wild, false,
+                            &Info);
```

Comment:
But doesn't `getINvalidCastPVCons` already constrain to wild? Why do you also need to call `constrainConsVarGeq` here?

Reply:
Maybe this is what `constraintAllCVarsToWild` was doing for all CVars, whereas this is just doing it for the outmost one?

Reply:
`getInvalidCastPVCons` constrains obtains a wild `PVConstraint`, but the expression being cast also needs to be made wild. This is done indirectly through the invalid cast constraint so that there is one edge going to WILD for an invalid cast, ensuring that the bad cast is only reported once by the root cause analysis.